prime-router/src/main/kotlin/cli/tests/Simulator.kt

```diff
@@ -369,6 +406,69 @@ class Simulator : CoolTest() {
 //            results += productionSimulation(environment, options)
 //            results += productionSimulation(environment, options)
 //        }
-        return teardown(results, elapsedTime)
+        return teardown(results, elapsedTime, startTime, options.asyncProcessMode)
+    }
+
+    /**
+     * Checks that at least the correct number of process records are in the Action table after running simulator.
+     * There may be more than expected if other tests occur at the same time, but there should not be fewer.
+     */
+    private suspend fun checkTimedResults(
+        expectedResults: Int,
+        afterDateTime: OffsetDateTime,
+        taskToCheck: TaskAction,
+        maxPollSecs: Int = 180,
+        pollSleepSecs: Int = 5
+    ): Boolean {
+        var resultsFound = 0
+
+        var timeElapsedSecs = 0
+        echo("Polling $taskToCheck records.  (Max poll time $maxPollSecs seconds)")
+        val actualTimeElapsedMillis = measureTimeMillis {
+            while (timeElapsedSecs <= maxPollSecs) {
+                if (outputToConsole) {
+                    for (i in 1..pollSleepSecs) {
+                        delay(1000)
+                        // Print out some contemplative dots to show we are waiting.
+                        print(".")
+                    }
+                    echo()
+                } else {
+                    delay(pollSleepSecs.toLong() * 1000)
+                }
+                timeElapsedSecs += pollSleepSecs
+
+                resultsFound = checkResultsQuery(afterDateTime, taskToCheck)
+
+                if (resultsFound >= expectedResults)
```

Comment:
Maybe print the numbers each time you poll, so you can watch them get bigger.  Makes testing so much more exciting!

Reply:
Updated